### PR TITLE
[RFC] Created a new method braceGlob to polyfill GLOB_BRACE

### PR DESCRIPTION
**Pull request by sheeep**
Discussion in #1057.

### CODE REVIEW

File: src/Resources/contao/library/Contao/Controller.php [hunk 1]
```diff
@@ -2341,4 +2343,30 @@ protected function createNewVersion($strTable, $intId)
 		$objVersions = new \Versions($strTable, $intId);
 		$objVersions->create();
 	}
+
+	protected static function braceGlob($resource)
+	{
+		if (false === strpos($resource, '/**/') && (defined('GLOB_BRACE') || false === strpos($resource, '{')))
+		{
+			return glob($resource, defined('GLOB_BRACE') ? GLOB_BRACE : 0);
```
Comment (Toflar) [inline src/Resources/contao/library/Contao/Controller.php]: The `defined()` check here is redundant :)

Comment (sheeep) [reply src/Resources/contao/library/Contao/Controller.php]: It's not.

Given that `false === strpos($resource, '/**/')` and `false === strpos($resource, '{'))` we still need that check. Even though the flag could be omitted IMO. This comes directly from [Symfonys `GlobalResource`](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Config/Resource/GlobResource.php#L98-L99).

Comment (Toflar) [reply src/Resources/contao/library/Contao/Controller.php]: Ah, true!

**Comment by leofeyer** [comment]
Merged into the correct branch in e98544b739d013a5019a8a5cad64ca96ae3087ba.

Thank you very much @sheeep. I did some profiling and `glob()` is about twice as fast, so good that we have both solutions now. 👍 

with-glob

without-glob

**Pull request merged**